android/src/main/java/com/clevertap/react/CleverTapCustomTemplates.kt

```diff
@@ -0,0 +1,66 @@
+package com.clevertap.react
+
+import android.content.Context
+import com.clevertap.android.sdk.CleverTapAPI
+import com.clevertap.android.sdk.inapp.customtemplates.CustomTemplateContext.FunctionContext
+import com.clevertap.android.sdk.inapp.customtemplates.CustomTemplateContext.TemplateContext
+import com.clevertap.android.sdk.inapp.customtemplates.CustomTemplateException
+import com.clevertap.android.sdk.inapp.customtemplates.FunctionPresenter
+import com.clevertap.android.sdk.inapp.customtemplates.TemplatePresenter
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.nio.charset.StandardCharsets
+
+object CleverTapCustomTemplates {
+
+    private val templatePresenter: TemplatePresenter = object : TemplatePresenter {
+        override fun onPresent(context: TemplateContext) {
+            CleverTapEventEmitter.emit(
+                CleverTapEvent.CLEVERTAP_CUSTOM_TEMPLATE_PRESENT, context.templateName
+            )
+        }
+
+        override fun onClose(context: TemplateContext) {
+            CleverTapEventEmitter.emit(
+                CleverTapEvent.CLEVERTAP_CUSTOM_TEMPLATE_CLOSE, context.templateName
+            )
+        }
+    }
+
+    private val functionPresenter = FunctionPresenter { context: FunctionContext ->
+        CleverTapEventEmitter.emit(
+            CleverTapEvent.CLEVERTAP_CUSTOM_FUNCTION_PRESENT, context.templateName
+        )
+    }
+
+    @JvmStatic
+    fun registerCustomTemplates(context: Context, vararg jsonAssets: String) {
+        for (jsonAsset in jsonAssets) {
+            val jsonDefinitions = readAsset(context, jsonAsset)
+            CleverTapAPI.registerCustomInAppTemplates(
+                jsonDefinitions, templatePresenter, functionPresenter
+            )
+        }
+    }
+
+    private fun readAsset(context: Context, asset: String): String {
```

Comment:
Will this throw a strict mode violation as to disk operation on the main thread?

Reply:
That is a good point. The file reading indeed happens on the main thread, but there is currently no easy way to do this work in another thread. This file must be parsed and templates mush be registered before the CleverTap SDK is initialized, which happens right after again on the main thread.

I enabled the StrictMode on the Example app and there were a lot of DiskRead violations happening, both originating in CleverTapSDK and in React Native's code itself. Interestingly registerCustomTemplates/readAsset were not reported, but it is possible that it was hidden by some other violation.

Reply:
`StrictMode` violations are not burning can can be ignored for now I guess provided that we are sure these won't lead to ANRs